addons/payment_adyen/utils.py

```diff
@@ -60,13 +60,16 @@ def format_partner_address(partner):
         'postalCode': partner.zip,
         # Fill in the address fields if the format is supported, or fallback to the raw address.
         'street': street_data.get('street_name', partner.street),
-        'houseNumberOrName': street_data.get('street_number'),
+        'houseNumberOrName': street_data.get('street_number') or '',
     }
     for key, value in address.items():
```

Comment:
```py
# If the incomplete addresses check is enabled, verify that no address detail is missing;
# if it is opted-out, return the address details as-is, because `split_street_with_params`
# sometimes fails to extract the street number, or non-mandatory address details
# (e.g., zip code) could be missing.
ignore_incomplete_adresses = partner.env['ir.config_paramer'].sudo().get_param('payment_adyen.ignore_incomplete_adress_details', False)
if not ignore_incomplete_adresses:
    for ...
       raise
 ```

Reply:
Plus, create the param record.